runtime/vm/CRIUHelpers.cpp

```diff
@@ -425,12 +434,41 @@ criuRestoreInitializeDump(J9VMThread *currentThread, void *userData, const char
 				*nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
 						J9NLS_VM_CRIU_RESTORE_INITIALIZE_DUMP_FAILED, NULL);
 				result = FALSE;
+			} else {
+				vm->checkpointState.flags |=  J9VM_CRIU_TRANSITION_TO_DEBUG_INTERPRETER;
 			}
 		}
 	}
 	return result;
 }
 
+/**
+ * An internal JVM checkpoint hook to check for the debug interpreter VMOPT after restore.
+ *
+ * @param[in] currentThread vmThread token
+ * @param[in] userData J9InternalHookRecord pointer
+ * @param[in/out] nlsMsgFormat an NLS message
+ *
+ * @return BOOLEAN TRUE
+ */
+static BOOLEAN
+criuRestoreCheckForDebugInterpreterRequest(J9VMThread *currentThread, void *userData, const char **nlsMsgFormat)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+
+	if (NULL != vm->checkpointState.restoreArgsList) {
+		J9VMInitArgs *restoreArgsList = vm->checkpointState.restoreArgsList;
+		IDATA debugOn = FIND_AND_CONSUME_ARG(restoreArgsList, EXACT_MATCH, VMOPT_XXDEBUGINTERPRETER, NULL);
+		IDATA debugOff = FIND_AND_CONSUME_ARG(restoreArgsList, EXACT_MATCH, VMOPT_XXNODEBUGINTERPRETER, NULL);
+		if (debugOn > debugOff) {
+			vm->checkpointState.flags |= J9VM_CRIU_TRANSITION_TO_DEBUG_INTERPRETER;
+		} else {
+			vm->checkpointState.flags &= ~(UDATA)J9VM_CRIU_TRANSITION_TO_DEBUG_INTERPRETER;
```

Comment:
In cases where this hook is run after `criuRestoreInitializeTrace` or `criuRestoreInitializeDump` and one of those sets:
`vm->checkpointState.flags |= J9VM_CRIU_TRANSITION_TO_DEBUG_INTERPRETER;`, do we want the presence of `-XX-DebugInterpreter` to override the transition request from `-Xtrace` or `-Xdump`? i.e. should this `else` case be here?

Reply:
> do we want the presence of -XX-DebugInterpreter to override the transition request from -Xtrace or -Xdump? i.e. should this else case be here?

No, if a hook has requested transition to debug interpreter then it means that we need it.

Reply:
The purpose of `-XX:-DebugInterpreter` is to negate the presence of `-XX:+DebugInterpreter` in the same set of cmdline args. Im not sure that is a usecase we need to support in restore args.